tendrl/node_agent/node_sync/sds_detect.py

```diff
@@ -35,112 +35,117 @@ def sync():
-                    try:
-                        integration_index_key = \
-                            "indexes/detected_cluster_id_to_integration_id/" \
-                            "%s" % sds_details['detected_cluster_id']
-                        _ptag = "provisioner/%s" % \
-                            NS.tendrl_context.integration_id
-                        _flow = "tendrl.flows." \
-                            "ExpandClusterWithDetectedPeers"
-                        if _ptag in NS.node_context.tags:
-                            dc = NS.tendrl.objects.DetectedCluster().load()
-                            if dc.detected_cluster_id and \
-                                dc.detected_cluster_id != sds_details.get(
-                                    'detected_cluster_id'):
-                                # Gluster peer list has changed
-                                integration_id = \
-                                    NS.tendrl_context.integration_id
-                                etcd_utils.write(
-                                    integration_index_key,
-                                    integration_id
-                                )
-                                # Let other nodes sync up with
-                                # integration_id
-                                time.sleep(10)
-                                params = {
-                                    'TendrlContext.integration_id':
-                                    integration_id,
-                                }
-                                payload = {
-                                    "tags": [_ptag],
-                                    "run": _flow,
-                                    "status": "new",
-                                    "parameters": params,
-                                    "type": "node"
-                                }
-                                _job_id = str(uuid.uuid4())
-                                _job = Job(
-                                    job_id=_job_id,
-                                    status="new",
-                                    payload=payload
-                                ).save()
-                                while True:
-                                    _job = _job.load()
-                                    if _job.status in ["finished",
-                                                       "failed"]:
-                                        break
-                            else:
-                                integration_id = str(uuid.uuid4())
-                                etcd_utils.write(integration_index_key,
-                                                 integration_id)
+                try:
+                    integration_index_key = \
+                        "indexes/detected_cluster_id_to_integration_id/" \
+                        "%s" % sds_details['detected_cluster_id']
+                    _ptag = "provisioner/%s" % \
```

Comment:
@shtripat @r0h4n my doubt is here we are checking provisioner tag but integration id is generated at line no 53 only 
https://github.com/Tendrl/node-agent/pull/732/files#diff-3dfeadf899a1fe4e98d3f57feef0b4daR53

Reply:
oh ok when dc_id is none we are creating uuid

Reply:
Yes I think first time integration_id would be None and so provisioner/None will not in tags list

Reply:
initially, it is None only, So here only we are creating integration_id